view/frontend/web/js/recobuy.js

```diff
@@ -35,56 +35,78 @@
 
 define([
     'nostojs',
-    'jquery',
-    'catalogAddToCart'
+    'jquery'
 ], function (nostojs, $) {
     'use strict';
 
-    //noinspection SpellCheckingInspection
-    var form = $('#nosto_addtocart_form').catalogAddToCart({});
     var Recobuy = {};
+
     Recobuy.addProductToCart = function (productId, element, quantity) {
         quantity = quantity || 1;
         var productData = {
-            "productId" : productId,
-            'skuId' : productId
+            productId : productId,
+            skuId : productId,
+            quantity: quantity
         };
-        Recobuy.addSkuToCart(productData, element, quantity);
+        return Recobuy.addSkuToCart(productData, element);
     };
 
     // Products must be and array of objects [{'productId': '123', 'skuId': '321'}, {...}]
     // skuId is optional for simple products.
     Recobuy.addMultipleProductsToCart = function (products, element) {
         if (Array.isArray(products)) {
```

Comment:
I don't think we need this guard. We don't validate the object anyways. Remove the check

Reply:
I can remove. It was altered here https://github.com/Nosto/nosto-magento2/pull/533/files#diff-f4a31733d3e7674b679d5cdfbff9eac23b3ba00542d9b44f98c1852d73625b9dR58